lib/integrations/samlidp/gcpworkforce.go

```diff
@@ -0,0 +1,212 @@
+
+	fmt.Println("\nCreating workforce pool: ", s.APIParams.PoolName)
+	poolFullName := fmt.Sprintf("locations/global/workforcePools/%s", s.APIParams.PoolName)
+	createPool := workforceService.Create(
+		"locations/global",
+		&iam.WorkforcePool{
+			Name:        poolFullName,
+			DisplayName: s.APIParams.PoolName,
+			Description: "Workforce pool created by Teleport",
+			Parent:      fmt.Sprintf("organizations/%s", s.APIParams.OrganizationID),
+		})
+	createPool.WorkforcePoolId(s.APIParams.PoolName)
+	resp, err := createPool.Do()
+	if err != nil {
+		// TODO(sshah): parse through error type for better error handling
+		return trace.Wrap(err)
+	}
+	fmt.Println("Pool created.")
+	if !resp.Done {
+		// 2 minutes timeout is semi-random decision, chosen based on the fact that
+		// when creating workforce pool from the GCP web console, it mentions
+		// that the operation could take up to 2 minutes.
+		pollCtx, cancel := context.WithTimeout(ctx, 120*time.Second)
+		defer cancel()
+
+		if err := waitForPoolStatus(pollCtx, workforceService, poolFullName, s.APIParams.PoolName); err != nil {
+			return trace.Wrap(err)
+		}
+		fmt.Printf("Pool %q is ready for use.\n", s.APIParams.PoolName)
+	}
+
+	fmt.Println("\nCreating workforce pool provider: ", s.APIParams.PoolProviderName)
+	metadata, err := fetchIdPMetadata(s.APIParams.SAMLIdPMetadataURL, s.HTTPClient)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	provider := &iam.WorkforcePoolProvider{
+		Description: "Workforce pool provider created by Teleport",
+		Name:        fmt.Sprintf("locations/global/workforcePools/%s/providers/%s", s.APIParams.PoolName, s.APIParams.PoolProviderName),
+		DisplayName: s.APIParams.PoolProviderName,
+		Saml: &iam.GoogleIamAdminV1WorkforcePoolProviderSaml{
+			IdpMetadataXml: metadata,
+		},
+		AttributeMapping: map[string]string{
+			"google.subject": "assertion.subject",
+			"google.groups":  "assertion.attributes.roles",
+		},
+	}
+
+	createProviderReq := workforceService.Providers.Create(poolFullName, provider)
+	createProviderReq.WorkforcePoolProviderId(s.APIParams.PoolProviderName)
+	createProviderResp, err := createProviderReq.Do()
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	if !createProviderResp.Done {
+		fmt.Printf("Pool provider %q is created but it may take upto a minute more for this provider to become available.\n\n", s.APIParams.PoolProviderName)
+		return nil
+	}
+	fmt.Println("Pool provider created.")
+	fmt.Printf("Pool provider %q is ready for use.\n", s.APIParams.PoolProviderName)
+	return nil
+}
+
+// waitForPoolStatus waits for pool to come online. Returns immediately if error code is other than
+// http.StatusForbidden or when context is canceled with timeout.
+func waitForPoolStatus(ctx context.Context, workforceService *iam.LocationsWorkforcePoolsService, poolName, poolDisplayName string) error {
+	fmt.Printf("Waiting for pool %q status to become available.\n", poolDisplayName)
+	ticker := time.NewTicker(1 * time.Second)
```

Comment:
> But having a 1 second ticker would be too noisy

Yaa I started with printing at every ticker but ended up with this one. 

>What about changing this to a 5 or 10 second ticker?

This works but unless we make the subsequent log `still waiting...` or some progress message, I doubt it adds much value. The wait time is also around 1 min on average so I would not further delay the ticker here.

Reply:
Just having a log line with `still waiting...` every 10s would be better, IMO.
Having no output indicating progress for 1 minute might led people thinking that it crashed or something.

Reply:
Fair enough on user expectation. In that case, I am updating the message "Waiting for pool status. It may take upto 2 minutes to become available." so user can expect that it takes some time.

Reply:
https://github.com/gravitational/teleport/pull/39820/commits/fa31ce655094db82bdb255a5d806e075d7fbf49f

I have updated ticker to 5 seconds just to be less aggressive. The message says now "Waiting for pool status. It may take upto 2 minutes for new pool to become available." I think this is now verbose enough to set expectation without further complicating `waitForPoolStatus` func. 

Feel free to suggest if you think otherwise :)